src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore
+      && !animal.hidden
+    ) {
+      animal.health -= 50;
+
+      Animal.alive = Animal.alive.filter(({
+        health,
+      }) => health > 0);
```

Comment:
```suggestion
      Animal.alive = Animal.alive
          .filter(({health}) => health > 0);
```

Reply:
If you have only one iteration method without a chain you can write it in the same string